### DISCUSSION TITLE
Performance issue accessing minio over LAN

### DISCUSSION THREAD

**Discussion by MichZipp** [minio/minio#12557] [Q&A] [answered]
I have a singleton installation of minio running on docker swarm and ran benchmarks to evaluate the performance.

Benchmark on the host A (on which minio is running):
Command used: `docker run minio/warp:latest mixed --host=<host_a_ip:host_a_port> --access-key=<access_key> --secret-key=<secret_key> --autoterm`
Results:
```
Mixed operations.
Operation: DELETE, 10%, Concurrency: 20, Duration: 4m59s.
 * Throughput: 36.54 obj/s

Operation: GET, 45%, Concurrency: 20, Duration: 4m59s.
 * Throughput: 1646.13 MiB/s, 164.61 obj/s

Operation: PUT, 15%, Concurrency: 20, Duration: 4m59s.
 * Throughput: 548.62 MiB/s, 54.86 obj/s

Operation: STAT, 30%, Concurrency: 20, Duration: 4m59s.
 * Throughput: 109.78 obj/s

Cluster Total: 2194.69 MiB/s, 365.79 obj/s over 4m59s.
```

Benchmark on host B (accessing host A via IP):
Command used: `docker run minio/warp:latest mixed --host=<host_a_ip:host_a_port> --access-key=<access_key> --secret-key=<secret_key> --autoterm`
Results:
```
Mixed operations.
Operation: DELETE, 10%, Concurrency: 20, Duration: 4m52s.
 * Throughput: 1.93 obj/s

Operation: GET, 45%, Concurrency: 20, Duration: 4m52s.
 * Throughput: 87.22 MiB/s, 8.72 obj/s

Operation: PUT, 15%, Concurrency: 20, Duration: 4m52s.
 * Throughput: 29.25 MiB/s, 2.93 obj/s

Operation: STAT, 30%, Concurrency: 20, Duration: 4m52s.
 * Throughput: 5.87 obj/s

Cluster Total: 116.27 MiB/s, 19.42 obj/s over 4m52s.
```

LAN benchmark between host A and host B via iPerf:
Command on host a: `iperf -s`
Command on host b: `iperf -c <host_a_ip>`
Results: 
```
[  3] local <host_b_ip> port 49174 connected with <host_a_ip> port 5001
[ ID] Interval       Transfer     Bandwidth
[  3]  0.0-10.0 sec  1.09 GBytes   940 Mbits/sec
```

The results show that accessing minio from another host is significantly slower than from the host minio is running on.
The LAN cannot be the issue since the iPerf benchmark shows satisfying results.

Anyone experienced the same issue and know what it could be?

Cheers!

**Comment by harshavardhana** [comment, marked as answer]
You LAN is 1Gib which is 120MiB/sec 

Your cluster total peaks at

Cluster Total: 116.27 MiB/s, 19.42 obj/s 

Which is equal to your network bandwidth.
